### ISSUE TITLE
Explore EK60 and EK80 data processing

### ISSUE THREAD

**Issue by simedroniraluca** [OceanStreamIO/echopype-scratchpad#14]
Comments and questions about EK data processing

**Comment by simedroniraluca** [comment]
@ruxandra-valcu

**Questions about Parameter Choices and `group_map` function in 'Basic echopype processing of EK60 files'  Notebook**

**Parameter Selection:** I noticed that range_sample_num and ping_num are set to 30 and 5 respectively in the ep.preprocess.remove_noise() function. Here's the specific snippet of code:

```python
#ds_SV type -> xarray.core.dataset.Dataset
#Remove background noise
ds_Sv_clean = ep.preprocess.remove_noise(    # obtain a denoised Sv dataset
   ds_Sv,             # calibrated Sv dataset
   range_sample_num=30,  # number of samples along the range_sample dimension for estimating noise
   ping_num=5,        # number of pings for estimating noise
)
```
I'm curious about how these particular values were chosen? 

**Krill Detection Thresholds**: Could you help me understand how the thresholds for the krill detection mask process were determined? What was the source of these values - Blackwell's thesis?

**`group_map`:** When using the ed.group_map command (where ed is an EchoData object), it seems to me that it's showing the standard netCDF format groups, not the ones specific to echopype. For example, I've noticed the 'beam2' group appearing in the output description, even though I was under the impression that echopype's convention was to use only the 'beam_1' group. Is my understanding correct or am I missing something here?

**Comment by simedroniraluca** [comment]
**Nice steps to process raw files using EK60**

Example of basic echopype processing of EK60 files by Ruxandra in this -> [GitHub repository](https://github.com/OceanStreamIO/echopype-scratchpad/blob/main/echopype-scratchpad/one_off_notebook/Basic%20echopype%20processing%20of%20EK60%20files.ipynb).

**Main echopype workflow steps - for each raw file:**

- **Access** file directly from S3 via `ep.open_raw()` to create a converted EchoData object in memory
- **Add global and platform attributes** to EchoData object
- **Export** to a local Zarr dataset 
- **Processing** - Sv computation and denoising:
    - Before computing the volume backscattering strength (Sv) using `ep.calibrate.compute_Sv()` in echopype, it is essential to identify the waveform mode, as additional parameters need to be set specifically for Broadband (BB) mode. 
    - Compute base Sv -> `ep.calibrate.compute_Sv()` -> (`import echopype as ep`)
    - Remove background noise -> `ep.preprocess.remove_noise()`
- Reduce data by computing Mean Volume Backscattering Strength (MVBS) -> `ep.preprocess.compute_MVBS()`
    - MVBS or cleaned Sv can be exported  to a local netcdf file
    - `xr.open_mfdataset()` -> (`import xarray as xr`) is used to open multiple netCDF files as a single xarray dataset
- Test for time reversals - a quality control test designed to identify instances in the echo sounder data where timestamps might be out of order -> See an example in the -> [GitHub repository](https://github.com/OSOceanAcoustics/echopype-examples/blob/main/notebooks/ms_PacificHake_EK60_cruisetracks.ipynb)
- Visualising echogram-> `epviz.create_echogram()` -> (`import echopype.visualize as epviz`)
- Particular example - Search for krill by creating a mask that keeps points with an Sv difference of 2-16 dB between the 120 kHz and 38 kHz channels.

A detailed walkthrough for EK60 data processing can be find also in the example by Raluca in this GitHub repository -> [GitHub repository](https://github.com/OceanStreamIO/echopype-scratchpad/blob/main/echopype-scratchpad/one_off_notebook/echopype_tour.ipynb)

**Comment by simedroniraluca** [comment]
@ruxandra-valcu Can you please help me understand where we can find information about whether the data collected with a specific echosounder uses Continuous Wave (CW) or Broadband (BB)? Can we identify this information in the netCDF file containing the echosounder data?

**Comment by ruxandra-valcu** [comment]
@simedroniraluca 

On parameter selection choices: it's just a proof of concept test to make sure that echopype functionality actually works, not the full optimized procedure :) (which will wait for when we have access to labeled training data and/or a domain expert who can confirm to us that what we think is krill, for example, actually is). For the denoise ones I just got them off a code snippet somewhere and had no reason to try other ones at this stage of the project's life, and for the krill detection mask I used some values suggested by the Korean paper Andrei linked us to.

On beam2: as per the echopype documentation, they're using beam2 for echosounders that store data in both complex and power format, when the echosounder is using both modes. EK80 is one of those, it can use both power (and then the file looks like an EK60 one) and complex mode (then, it'll have beam_r and beam_i populated instead of the power and angle ones). So if you have an ek80 file with a single beam1 dataset it means it's using a consistent mode, and you realise which one by looking at which fields it has data in. And then if it's power it means it's using CW mode and if it's broadband it means it's using BB mode. 

These sound like great stuff to add to the notebook btw, make it more self-explanatory :)
